Source/JavaScriptCore/wasm/js/JSWebAssemblyInstance.h

```diff
@@ -280,12 +280,12 @@ class JSWebAssemblyInstance final : public JSNonFinalObject {
 
     static_assert(sizeof(WasmOrJSImportableFunctionCallLinkInfo) == WTF::roundUpToMultipleOf<sizeof(uint64_t)>(sizeof(WasmOrJSImportableFunctionCallLinkInfo)), "We rely on this for the alignment to be correct");
     static constexpr size_t offsetOfTablePtr(unsigned numImportFunctions, unsigned i) { return offsetOfTail() + sizeof(WasmOrJSImportableFunctionCallLinkInfo) * numImportFunctions + sizeof(Wasm::Table*) * i; }
-    static constexpr size_t offsetOfGlobalPtr(unsigned numImportFunctions, unsigned numTables, unsigned i) { return roundUpToMultipleOf<sizeof(Wasm::Global::Value)>(offsetOfTablePtr(numImportFunctions, numTables)) + sizeof(Wasm::Global::Value) * i; }
-    static constexpr size_t offsetOfGCObjectStructure(unsigned numImportFunctions, unsigned numTables, unsigned numGlobals, unsigned i) { return offsetOfGlobalPtr(numImportFunctions, numTables, numGlobals) + sizeof(WriteBarrier<WebAssemblyGCStructure>) * i; }
-    WriteBarrier<WebAssemblyGCStructure>& gcObjectStructure(unsigned numImportFunctions, unsigned numTables, unsigned numGlobals, unsigned i) { return *std::bit_cast<WriteBarrier<WebAssemblyGCStructure>*>(reinterpret_cast<char*>(this) + offsetOfGCObjectStructure(numImportFunctions, numTables, numGlobals, i)); }
-    WriteBarrier<WebAssemblyGCStructure>& gcObjectStructure(unsigned typeIndex) { return gcObjectStructure(numImportFunctions(), moduleInformation().tableCount(), moduleInformation().globalCount(), typeIndex); }
+    static constexpr size_t offsetOfGlobalPtr(unsigned numImportFunctions, unsigned numTables, unsigned i) { return roundUpToMultipleOf<alignof(Wasm::Global::Value)>(offsetOfTablePtr(numImportFunctions, numTables)) + sizeof(Wasm::Global::Value) * i; }
+    static constexpr size_t offsetOfGCObjectStructureID(unsigned numImportFunctions, unsigned numTables, unsigned numGlobals, unsigned i) { return WTF::roundUpToMultipleOf<alignof(WriteBarrierStructureID)>(offsetOfGlobalPtr(numImportFunctions, numTables, numGlobals)) + sizeof(WriteBarrierStructureID) * i; }
+    WriteBarrierStructureID& gcObjectStructureID(unsigned numImportFunctions, unsigned numTables, unsigned numGlobals, unsigned i) { return *std::bit_cast<WriteBarrierStructureID*>(reinterpret_cast<char*>(this) + offsetOfGCObjectStructureID(numImportFunctions, numTables, numGlobals, i)); }
+    WriteBarrierStructureID& gcObjectStructureID(unsigned typeIndex) { return gcObjectStructureID(numImportFunctions(), moduleInformation().tableCount(), moduleInformation().globalCount(), typeIndex); }
```

Comment:
Why not have both? Then you don't need most of the `jsCast<WebAssemblyGCStructure*>(instance->gcObjectStructureID(typeIndex).get())` changes.